transcom/mymove
# Int b 19808 back end sc ability to edit departure date and aoa

## [B-19808](https://www13.v1host.com/USTRANSCOM38/assetdetail.v1?number=B-19808)

## Summary

This is Backend work for SC ability to edit `Actual move start date` and `Advance received information while on the Review Documents page`. Also fixed the bug where Advance received should not show in the UI when Advance requested is null.

### How to test

1. Create a move with a PPM shipment.
2. As a SC -> Counsel/Submit the move.
3. As a customer -> Fill out all the information in the PPM Documents page and submit.
4. As a Closeout SC -> go to the move and click Review Documents to review.
5. Open up inspector tool/developer tool and go to the network tab.
6. Refresh the page and look for `ppmShipment` object within `mto_shipments`.
7. Find and take note of `actualMoveDate`, `hasReceivedAdvance`, and `advanceAmountReceived`
8. Open Postman and find the PATCH in this dir:
![image](https://github.com/transcom/mymove/assets/146856854/49587759-e63b-4930-be59-c7f3d79d4001)
9. Fill out needed variables in the Params, Headers Tab and the cookies.
10. In the Body, use `raw` and delete all the pre-existing object data and use this object as a reference to update data needed:
```
{
  "ppmShipment": {
    "actualMoveDate": "2020-03-12",
    "advanceAmountReceived": 111111,
    "hasReceivedAdvance": true
  }
}
```
11. After success, refresh the Review Documents page and check if the values were updated.


## Screenshots
![image](https://github.com/transcom/mymove/assets/146856854/2a768d45-b54f-4513-bab1-884fce31439c)


![image](https://github.com/transcom/mymove/assets/146856854/65179a18-5c50-445d-a742-bd14abc76308)
